cmd/kube-apiserver/app/server.go

```diff
@@ -210,9 +212,16 @@ func Run(s *options.APIServer) error {
 				installSSH = instances.AddSSHKeyToAllInstances
 			}
 		}
-
+		if ports.KubeletReadOnlyPort <= 0 {
+			glog.Fatalf("Must enable kubelet read-only port if proxy ssh-tunneling is specified.")
+		}
 		// Set up the tunneler
-		tunneler = master.NewSSHTunneler(s.SSHUser, s.SSHKeyfile, installSSH)
+		healthCheckPath := &url.URL{
+			Scheme: "http",
+			Host:   net.JoinHostPort("127.0.0.1", strconv.Itoa(ports.KubeletReadOnlyPort)),
```

Comment:
It should work if kubelet-in-a-container is in the host network namespace. And I'll look into using the secure port. There's gotta be a TLS config that I can build up in this context...